src/openrct2/windows/loadsave.c

```diff
@@ -812,6 +821,10 @@ static void window_loadsave_select(rct_window *w, const char *path)
 			window_loadsave_invoke_callback(MODAL_RESULT_FAIL, path);
 		}
 	}
+	case (LOADSAVETYPE_LOAD | LOADSAVETYPE_IMAGE):
+		window_close_by_class(WC_LOADSAVE);
+		window_loadsave_invoke_callback(MODAL_RESULT_OK, path);
```

Comment:
How is this reading now-deallocated memory? `path` is an argument, and the callback is stored in a global static variable, not in the window. I can move the closing to the callback instead, but then you are not guaranteed that the window will not be used after the callback has fired.